spring-cloud-context/src/main/java/org/springframework/cloud/autoconfigure/RefreshEndpointAutoConfiguration.java

```diff
@@ -45,6 +45,7 @@
  */
 @Configuration
 @ConditionalOnClass(Endpoint.class)
+@ConditionalOnProperty(value = "endpoints.enabled", matchIfMissing = true)
```

Comment:
I don't think this is required. See https://github.com/spring-projects/spring-boot/blob/master/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/condition/OnEnabledEndpointCondition.java#L67-L75

`@ConditionalOnEnabledEndpoint` should take it into account.